client/src/controllers/DrilldownController.ts

```diff
@@ -0,0 +1,116 @@
+    const swapEvent = e as CustomEvent<{ requestUrl: string }>;
+    if ((e.target as HTMLElement)?.id === 'listing-results') {
+      const params = new URLSearchParams(
+        swapEvent.detail?.requestUrl.split('?')[1],
+      );
+      const filteredParams = new URLSearchParams();
+      params.forEach((value, key) => {
+        if (value.trim() !== '') {
+          // Check if the value is not empty after trimming white space
+          filteredParams.append(key, value);
+        }
+      });
+      const queryString = '?' + filteredParams.toString();
+      this.updateToggleCounts();
+      window.history.replaceState(null, '', queryString);
+    }
+  }
+
+  open(e: MouseEvent) {
+    const toggle = (e.target as HTMLElement)?.closest(
+      'button',
+    ) as HTMLButtonElement;
+    this.activeSubmenuValue = toggle.getAttribute('aria-controls') || '';
+  }
+
+  close() {
+    this.activeSubmenuValue = '';
+    this.updateToggleCounts();
+  }
+
+  activeSubmenuValueChanged(activeSubmenu: string, prevActiveSubmenu?: string) {
+    if (prevActiveSubmenu) {
+      const toggle = document.querySelector(
+        `[aria-controls="${prevActiveSubmenu}"]`,
+      ) as HTMLButtonElement;
+      this.toggle(false, toggle);
+    }
+
+    if (activeSubmenu) {
+      const toggle = document.querySelector(
+        `[aria-controls="${activeSubmenu}"]`,
+      ) as HTMLButtonElement;
+      this.toggle(true, toggle);
+    }
+  }
+
+  toggle(expanded: boolean, toggle: HTMLButtonElement) {
+    const controls = toggle.getAttribute('aria-controls');
+    const content = this.element.querySelector<HTMLElement>(`#${controls}`);
+    if (!content) {
+      return;
+    }
+    toggle.setAttribute('aria-expanded', expanded.toString());
+    content.hidden = !expanded;
+    this.menuTarget.hidden = expanded;
+    this.element.classList.toggle('w-drilldown--active', expanded);
+
+    if (expanded) {
+      content.focus();
+    } else {
+      toggle.focus();
+    }
+  }
+
+  /**
+   * Placeholder function until this is more correctly set up with the backend.
+   */
+  updateToggleCounts() {
+    let sum = 0;
+
+    this.toggleTargets.forEach((toggle) => {
+      const content = this.element.querySelector<HTMLElement>(
+        `#${toggle.getAttribute('aria-controls')}`,
+      );
+      const counter = toggle.querySelector<HTMLElement>('.w-drilldown__count');
+      if (!content || !counter) {
+        return;
+      }
+      // Hack to detect fields with a non-default value.
+      const nbActiveFields = content.querySelectorAll(
+        '[type="checkbox"]:checked, [type="radio"]:checked:not([id$="_0"]), option:checked:not(:first-child), input:not([type="checkbox"], [type="radio"]):not(:placeholder-shown)',
+      ).length;
+      counter.hidden = nbActiveFields === 0;
+      counter.textContent = nbActiveFields.toString();
+      sum += nbActiveFields;
+    });
+
+    const sumCounter = this.element.querySelector<HTMLElement>(
+      '.w-drilldown__count',
```

Comment:
If possible, it would be good to use a Stimulus target instead of a selector with a class for finding the count element.

Reply:
This is a PoC method, the count isn’t even meant to be calculated client-side.

Reply:
Ahh ok.